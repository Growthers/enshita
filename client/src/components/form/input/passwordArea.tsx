import { useState } from "react";
import styles from "~/styles/components/form/input/textArea.module.scss";
import { Icon } from "@iconify/react";
import { PasswordProps } from "~/components/form/input/type/model";

const Password = <T extends string>({
  label,
  discription,
  register,
  itsError,
}: PasswordProps<T>): JSX.Element => {
  const [IsShow, setShow] = useState(true);

  return (
    <div className={styles["enshita-textarea-form-body"]}>
      <label htmlFor={label}>
        <p>
          <small>{discription}</small>
        </p>
        <div className={styles["enshita-textarea-textarea"]}>
          <div className={styles["enshita-textarea-icon"]}>
            <Icon icon="bxs:lock-alt" width="15" height="15" />
          </div>
          <input
            type={IsShow ? "password" : "text"}
            className={styles["enshita-textarea-input-password"]}
            {...register}
          />
          <button
            type="button"
            className={styles["enshita-textarea-icon"]}
            onClick={() => {
              setShow(!IsShow);
            }}
          >
            {IsShow && (
              <Icon icon="ant-design:eye-filled" width="15" height="15" />
            )}
            {!IsShow && (
              <Icon
                icon="ant-design:eye-invisible-filled"
                width="15"
                height="15"
              />
            )}
          </button>
        </div>
        <div className={styles["enshita-textarea-error"]}>
          {itsError && (
            <p>
              <small>{`${
                itsError.ref?.value === ""
                  ? "this field is required"
                  : itsError.message
              }`}</small>
            </p>
          )}
        </div>
      </label>
    </div>
  );
};

export default Password;
