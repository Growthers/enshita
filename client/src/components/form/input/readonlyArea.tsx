import styles from "~/styles/components/form/input/textArea.module.scss";
import { Icon } from "@iconify/react";
import { NormalProps } from "~/components/form/input/type/model";

const ReadOnly = <T extends string>({
  label,
  discription,
  icon,
  register,
  itsError,
}: NormalProps<T>): JSX.Element => (
  <div className={styles["enshita-textarea-form-body"]}>
    <label htmlFor={label}>
      <p>
        <small>{discription}</small>
      </p>
      <div className={styles["enshita-textarea-textarea"]}>
        <div className={styles["enshita-textarea-icon"]}>
          <Icon icon={icon} width="15" height="15" />
        </div>
        <input
          className={styles["enshita-textarea-input-pasword"]}
          {...register}
          readOnly
        />
        <div className={styles["enshita-textarea-icon"]}>
          <Icon icon="carbon:edit-off" width="15" height="15" />
        </div>
      </div>
      <div className={styles["enshita-textarea-error"]}>
        {itsError && (
          <p>
            <small>{`${
              itsError.ref?.value === ""
                ? "this field is required"
                : itsError.message
            }`}</small>
          </p>
        )}
      </div>
    </label>
  </div>
);

export default ReadOnly;
