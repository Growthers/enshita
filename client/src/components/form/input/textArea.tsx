import styles from "~/styles/components/form/input/textArea.module.scss";
import { Icon } from "@iconify/react";
import { NormalProps } from "~/components/form/input/type/model";

const Normal = <T extends string>({
  label,
  discription,
  icon,
  register,
  itsError,
}: NormalProps<T>): JSX.Element => (
  <div className={styles["enshita-textarea-form-body"]}>
    <label htmlFor={label}>
      <p>
        <small>{discription}</small>
      </p>
      <div className={styles["enshita-textarea-textarea"]}>
        <div className={styles["enshita-textarea-icon"]}>
          <Icon icon={icon} width="15" height="15" />
        </div>
        <input
          className={styles["enshita-textarea-input-normal"]}
          {...register}
        />
      </div>
      <div className={styles["enshita-textarea-error"]}>
        {itsError && (
          <p>
            <small>{itsError.message}</small>
          </p>
        )}
      </div>
    </label>
  </div>
);

export default Normal;
