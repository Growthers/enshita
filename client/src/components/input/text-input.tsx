import { forwardRef } from "react";
import styles from "~/styles/components/input/input.module.scss";
import type { TextInputProperties } from "./type/model";

// eslint-disable-next-line react/display-name
const TextInput = forwardRef<HTMLInputElement, TextInputProperties>(
  ({ inputStyles, error, ...props }, ref) => (
    <>
      <input ref={ref} className={inputStyles} {...props} />
      {error && (
        <p aria-label="error message" className={styles["enshita-input-error"]}>
          {error.message}
        </p>
      )}
    </>
  ),
);

export { TextInput };
