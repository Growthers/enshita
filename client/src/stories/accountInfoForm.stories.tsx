// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
