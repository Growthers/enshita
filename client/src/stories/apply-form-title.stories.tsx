import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
