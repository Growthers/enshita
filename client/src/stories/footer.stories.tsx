// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
