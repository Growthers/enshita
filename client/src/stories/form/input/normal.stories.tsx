// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Normal from "~/components/account-info-form/input/textArea";

type T = typeof Normal;
type Story = ComponentStoryObj<T>;

export default {
  component: Normal,
  args: {
    label: "mail",
    discription: "mail",
    icon: "ci:mail",
  },
} as ComponentMeta<T>;

export const Default: Story = {};
