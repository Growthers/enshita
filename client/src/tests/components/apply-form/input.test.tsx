import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "../../../stories/apply-form/input.stories";

const { Default, Valid } = composeStories(stories);

describe("(components) text input with using react-hook-form", () => {
  test("label should be name in English", () => {
    const { getByLabelText } = render(<Default />);
    expect(getByLabelText("名前")).toBeInTheDocument();
  });
  test("placeholder should be name in English", () => {
    const { getByPlaceholderText } = render(<Default />);
    expect(getByPlaceholderText("名前")).toBeInTheDocument();
  });
  test("(valid) input is correctly entered", async () => {
    const { container, getByLabelText } = render(<Valid />);
    await Valid.play({ canvasElement: container });
    const input = getByLabelText("名前") as HTMLInputElement;
    expect(input.value).toEqual("Rintaro Itokawa");
  });
  test("test Snap Shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
