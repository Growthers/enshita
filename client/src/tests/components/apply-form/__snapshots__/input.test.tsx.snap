// Vitest Snapshot v1

exports[`(components) text input with using react-hook-form > take snapshot 1`] = `
<div>
  <form>
    <section
      class="_enshita-apply-form-input-box_cd2e7a"
    >
      <label
        class="_enshita-apply-form-input-label_cd2e7a"
        for="name"
      >
        名前
      </label>
      <p
        class="_enshita-apply-form-description_cd2e7a"
      >
        名前を入力してください
      </p>
      <input
        aria-invalid="false"
        aria-label="name"
        class="_enshita-apply-form-input_cd2e7a"
        id="name"
        name="name"
        placeholder="名前"
        value=""
      />
    </section>
  </form>
</div>
`;
